Integrations/integration-Tufin.yml

```diff
@@ -0,0 +1,334 @@
+            return_error('Unknown Output Returned')
+        # Send back to Demisto inside function
+        return entry
+
+    def path_finder_command():
+        """ Sample query: querystring = {'src':'10.80.80.0','dst':'172.16.200.80','service':'tcp:22','includeIncompletePaths':'true'} """
+        # Build the query from user input
+        querystring = {
+            'src': demisto.args()['source'],
+            'dst': demisto.args()['destination'],
+            'service': demisto.args().get('service', 'Any'),
+            'includeIncompletePaths': 'true'
+        }
+        e = path_finder(querystring)
+        demisto.results(e)
+
+    def path_finder_image(querystring):
+        entry = {
+            'Type': entryTypes['note'],
+            'Contents': '',
+            'ContentsFormat': formats['text'],
+            'ReadableContentsFormat': formats['markdown'],
+            'HumanReadable': ''
+        }
+
+        try:
+            headers = {'accept': 'image/png','content-type': 'application/json','cache-control': 'no-cache'}
+            img = tos_request('/securetrack/api/topology/path_image', querystring, headers)
+            # simple check if we have an image or error message.
+            if len(img) > 100:
+                # Send back to Demisto inside function
+                return fileResult('topo.png', img, entryTypes['image'])
+            else:
+                entry['HumanReadable'] = 'No Valid Path Found'
+                entry['Contents'] = 'No Valid Path Found'
+                # Send back to Demisto inside function
+                return entry
+        except:
+            return_error('Error Running Query')
+
+    def path_finder_image_command():
+        """ Sample query: querystring = {'src':'10.80.80.0','dst':'172.16.200.80','service':'tcp:80','includeIncompletePaths':'true','displayBlockedStatus':'true'} """
+        querystring = {
+            'src': demisto.args()['source'],
+            'dst': demisto.args()['destination'],
+            'service': demisto.args().get('service', 'Any'),
+            'includeIncompletePaths': 'true',
+            'displayBlockedStatus': 'true'
+        }
+        e = path_finder_image(querystring)
+        demisto.results(e)
+
+    def device_name(devices, device_id):
+        return [e['name'] + ' ({} {})'.format(e['vendor'], e['model']) for e in devices if int(e['id']) == int(device_id)][0]
+
+    def object_lookup(querystring):
+        entry = {
+            'Type': entryTypes['note'],
+            'Contents': '',
+            'ContentsFormat': formats['json'],
+            'ReadableContentsFormat': formats['markdown'],
+            'HumanReadable': ''
+        }
+
+        return_json = {'objects': []}
+
+        o = tos_request('/securetrack/api/network_objects/search', querystring)
+
+        # Validate result
+        try:
+            total = int(o['network_objects']['count'])
+        except KeyError:
+            total = 0
+
+        if total > 0:
```

Comment:
All this logic should be in the `command` function

Reply:
This is validating the response which can't happen until the request.